eogrow/pipelines/export_maps.py

```diff
@@ -235,51 +212,142 @@ def _prepare_files(self, geotiff_paths: List[str], output_file: str) -> Tuple[FS
+        for tiff_path in geotiff_paths:
+            tiff_sys_path = filesystem.getsyspath(tiff_path)
 
-    def _load_timestamp(self, eopatch_name: str) -> List[dt.datetime]:
+            jobs = []
+            for i, time in enumerate(timestamp):
+                extraction_path = get_extraction_path(tiff_path, time)
+                bands = range(i * num_bands, (i + 1) * num_bands)
+                jobs.append(SplitTiffsJob(tiff_sys_path, bands, filesystem.getsyspath(extraction_path)))
+
+                geotiffs_per_time[time].append(extraction_path)
+
+            grouped_split_jobs.append(jobs)
+
+        parallelize(self._execute_split_jobs, grouped_split_jobs, workers=None)
+        parallelize(filesystem.remove, geotiff_paths, workers=None, multiprocess=False)
+
+        return geotiffs_per_time
+
+    def _load_bands_and_timestamp(self, eopatch_name: str) -> Tuple[int, List[dt.datetime]]:
+        """Loads an eopatch to get information about number of bands and the timestamp."""
         path = fs.path.join(self.storage.get_folder(self.config.input_folder_key), eopatch_name)
-        patch = EOPatch.load(path, FeatureType.TIMESTAMP, filesystem=self.storage.filesystem)
-        return patch.timestamp
-
-    def _split_temporally(
-        self, filesystem: FS, map_path: str, timestamp: List[dt.datetime], output_folder: str, crs: CRS
-    ) -> List[Tuple[str, Optional[dt.datetime]]]:
-        """Splits the merged tiff into multiple tiffs, one per timestamp."""
-        filesystem.makedirs(output_folder, recreate=True)  # in case we use a temporary filesystem
-
-        with filesystem.openbin(map_path) as file_handle:
-            with rasterio.open(file_handle) as map_src:
-                num_bands = map_src.count // len(timestamp)
-
-        LOGGER.info("Splitting merged tiff into per-timestamp tiff files.")
-        outputs: List[Tuple[str, Optional[dt.datetime]]] = []
-        for i, time in tqdm(enumerate(timestamp), desc="Spliting per timestamp", total=len(timestamp)):
-            name = self.get_geotiff_name(self.MERGED_MAP_NAME, crs, time)
-            extraction_path = fs.path.join(output_folder, name)
-            bands = range(i * num_bands, (i + 1) * num_bands)
-            extract_bands(filesystem.getsyspath(map_path), filesystem.getsyspath(extraction_path), bands, quiet=True)
-            outputs.append((extraction_path, time))
-
-        filesystem.remove(map_path)
-        return outputs
-
-    def _finalize_output_files(
-        self, filesystem: FS, output_paths: List[Tuple[str, Optional[dt.datetime]]], output_folder: str
+        patch = EOPatch.load(path, (FeatureType.TIMESTAMP, self.config.feature), filesystem=self.storage.filesystem)
+        if self.config.band_indices is not None:
+            return len(self.config.band_indices), patch.timestamp
+        return patch[self.config.feature].shape[-1], patch.timestamp
```

Comment:
- the loading should be lazy
- the if-return could be placed at the start

Reply:
we need the timestamps, not sure how to move it to an earlier location

Reply:
good point, ignore my first point.

what about reading the num_features with `gdalinfo`? are you against that idea?

Reply:
not that against it, just cant yet see the benefit of that over what is currently done
what I see is that now we use one approach to get two informations, but then we would have two approaches for two informations. What is in your opinion the main benefit of `gdalinfo`? :thinking:

Reply:
My suggestion was based on the fact that you only need to read the total number of bands from the tiff with gdal, which is quick, but here you need to read the full eopatch feature (which can be large) to get the size of the band dimension, but I'm not sure about the actual comparison of processing time.

I don't have any preference, just wanted to understand the reasoning, since the gdal way was used before the change, if I understand correctly.

Reply:
before I used `rasterio` for that. And if it came down to checking the tiffs, I'd probably use the same approach (no subprocess call, no need to parse the output of gdalinfo).

This is done once per CRS (so even for large zones like half of africa this is done 10 times in the run of the full pipeline, where this features was already loaded 1000 times to export the tiffs) so I consider this negligible when it comes to efficiency. But on the other hand the rasterio implementation shouldn't need a lot of lines of code (probably about 3 or 4 extra), so I could switch to it if you feel like this slowdown might be considerable.

Reply:
> so I consider this negligible when it comes to efficiency

OK, I'm convinced

Reply:
so to reiterate, the current approach was chosen because it requires two additional lines of code and no additional mechanic, and I consider the slowdown to be acceptable.